hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -3779,19 +3779,81 @@ private String onRegionSplitReverted(ServerName sn,
     }
     return null;
   }
+  
+  /* BEGIN changes so that regions during split and merge using FavoredNodes */
+  public void assignDaughterRegions(
```

Comment:
These changes I need to do a compare with trunk and see why you wanted this. But if you can explain the motive and the issue it would be easier to understand. And as I already said if you feel there are issues in 1.2.x branch and you want to make it work as in trunk you can definitely raise a backport issue (considering the fix is applicable and possible to do as in trunk)